controls/src/main/java/org/tweetwallfx/controls/stepengine/Step.java

```diff
@@ -24,18 +24,57 @@
 package org.tweetwallfx.controls.stepengine;
 
 import java.time.Duration;
+import org.tweetwallfx.controls.dataprovider.DataProvider;
+import org.tweetwallfx.controls.stepengine.StepEngine.MachineContext;
 
 /**
- *
  * @author Jörg Michelberger
  */
 public interface Step {
-    void initStep(StepEngine.MachineContext context);
-    boolean shouldSkip(StepEngine.MachineContext context);
-    void prepareStep(StepEngine.MachineContext context);
-    void doStep(StepEngine.MachineContext context);
-    void leaveStep(StepEngine.MachineContext context);
-    Duration preferredStepDuration(StepEngine.MachineContext context);
-    String getName();
-    boolean requiresPlatformThread();
+
+    default void initStep(final MachineContext context) {
+    }
+
+    default boolean shouldSkip(final MachineContext context) {
+        return false;
+    }
+
+    default void prepareStep(final MachineContext context) {
+    }
+
+    void doStep(final MachineContext context);
+
+    default void leaveStep(final MachineContext context) {
+    }
+
+    Duration preferredStepDuration(final MachineContext context);

```

Comment:
Should preferredStepDuration have a default implementation, e.g Duration.ZERO?

Reply:
okay for me. @jmichelberger what do you think?